src/highdicom/sr/templates.py

```diff
@@ -419,10 +528,17 @@ def _contains_scoord_items(
     )
     for item in matched_items:
         if graphic_type is not None:
-            if item.GraphicType == graphic_type.value:
-                return True
-        else:
-            return True
+            if item.GraphicType != graphic_type.value:
+                continue
+        if source_image_sop_instance_uid is not None:
+            if not _contains_image_items(
+                item,
+                name=codes.DCM.OriginalSource,
+                referenced_sop_instance_uid=source_image_sop_instance_uid,
+                relationship_type=RelationshipTypeValues.SELECTED_FROM
+            ):
+                continue
+        return True
```

Comment:
Is this the desired behaviour?

Reply:
This method is actually no longer needed and was removed in bd0f5d3